blueocean-rest-impl/src/main/java/io/jenkins/blueocean/service/embedded/rest/MultiBranchPipelineQueueContainer.java

```diff
@@ -36,13 +36,19 @@ public BlueQueueItem get(String name) {
             if(item != null){
                 PipelineImpl pipeline = (PipelineImpl) multiBranchPipeline.getBranches().get(item.task.getOwnerTask().getName());
                 if(pipeline != null) {
-                    int runId=0;
+
                     if(item.task instanceof ExecutorStepExecution.PlaceholderTask) {
                         ExecutorStepExecution.PlaceholderTask task = (ExecutorStepExecution.PlaceholderTask) item.task;
-                        runId = task.run().getNumber();
+                        int runNumber;
+                        if(task.run() == null){
+                            runNumber = pipeline.job.getNextBuildNumber();
```

Comment:
@imeredith I could do that but thats unnecessary expense (to iterate thru all queued items) and even with that you never know if you got the build number in sync with what jenkins thinks it is. Will be great if your replace PR can get us such info.